### ISSUE TITLE
[Feature] Allow blacklisted items to show amount for current character through toggle/option?

### ISSUE THREAD

**Issue by Raenore** [Xruptor/BagSync#353]
Hello there,

I've switched over to BagSync mainly because I wanted to see which char had item X, Y, Z.
It does this job fantastically and without requiring too many resources.

I also use ElvUI and that allows me to have a "Bag: X" number, indicating amount of items in my bag on the current char.
This is something that BagSync emulates with the Bag: x and Total: x.

However I was curious if it's possible to still show how many items are in your CURRENT bag even if an item is blacklisted?
I personally blacklist my heirlooms, but sometimes I do want to know if I do actually have one or two (like dual-wielding weapons) in my bag.

Now, I can turn this on through ElvUI and it fixes that. But then for non-blacklisted items it now shows how many I have in my bags twice. (Once from BagSync, once from ElvUI). Which is a bit silly unfortunately.

So I figured I'd ask if it's possible for a means to keep showing how many of an item you have in your current bag without it being over your whole account. As I notice that even with blacklisted items it still shows "ItemTypes" and "Expansion".

| Blacklisted   | Blacklisted (ElvUI) | Non-Blacklisted (ElvUI + BGS) |
| :-----------: | :-----------: | :-----------: |
| ![image](https://github.com/user-attachments/assets/3a72955d-f6b2-4145-87f6-877879614c58) | ![image](https://github.com/user-attachments/assets/27a3f443-1f9a-46e0-a586-3875279684bd) | ![image](https://github.com/user-attachments/assets/da198aaa-25e7-45a0-984a-dc85addd908d) |

Thanks in advance, lovely and amazing addon!

**Comment by Xruptor** [comment]
Blacklisting an item will only prevent it being from tallied up in any counts.  It doesn't actually apply to any of the other tooltip options available for additional info  (such has itemtypes, expansions, itemid, etc...).  That's totally separate to the tally count and it's more of optional flavor info a person can turn on. 

I'm still a little confused as to what exactly you want?  I'll reread everything but I got lost in the ElvUI settings view vrs the BagSync stuff.  The two addons don't really talk to each other so requests like these usually are very specific.

The reason it shows 2 in the last picture is because you have 2 characters that have that item.  It's counting everything on the account and not just the current character.  Whereas ElvUI is only counting stuff on the current character and since it was blacklisted it's showing 1.  The difference between the two is that ElvUI only does the current character whereas BagSync does all your characters.

There is an option to display item totals for your current character ONLY in BagSync.  But I really don't recommend that as it affects all items and not just blacklisted ones.  Hope that helps! 💚

**Comment by Raenore** [comment]
Hello there,

The question is mostly this:

Once you blacklist an item, BagSync no longer shows item totals. It only seems to still show ItemType and Expansion (in my case).
I'd like to know if there is an option (and if not, if it can be made) where blacklisted items still show the current character's total by its own.

That way I don't have to enable ElvUI's "Bag Totals" like I did in my second image.

**Comment by Xruptor** [comment]
> Once you blacklist an item, BagSync no longer shows item totals. It only seems to still show ItemType and Expansion (in my case).
> 

This is by design.  I mentioned that above.  The blacklist only affects the tally counts.

> I'd like to know if there is an option (and if not, if it can be made) where blacklisted items still show the current character's total by its own.
> 

I mean that kind of defeats the whole purpose of using a Blacklist in the first place.  I'll have to double check but when I made the system I didn't have something like this in mind.  Partially because it's parsing all your characters and tallying up all the counts.  If an item is on the blacklist by itemID it just ignores it in general.  So to have it check the Blacklist and then ignore it if it's the current character and only if a particular option is on is somewhat unusual.  I'd have to give it some thought as there is an option to only show the current player tooltip right now.  What you are asking would have to have additional coding.  I can't promise you I'll incorporate it, but I do take everything into consideration.  This seems like a very unique situation.

**Comment by Raenore** [comment]
No worries at all if it's not possible/out of scope for the addon!

I just like knowing at all times how many of an item I have in my bags.

Like mentioned before: I can turn that on through ElvUI but then it will display the count twice on non-blacklisted items (both by ElvUI and BagSync). Which is something I wanted to avoid. 

But truly, no worries if it isn't possible/planned. I figured I would ask as I saw blacklisted items still showed some BagSync info on the tooltip, so I wrongly guessed the char-only item count was held somewhere as well.

**Comment by Xruptor** [comment]
I added a new option to the Display category to show blacklisted item counts for the current player only.  **Please let me know if this works for you and it's what you wanted.**

NOTE:  I added a tag [Blacklist] to remind you that it's been blacklisted.
[BagSync.zip](https://github.com/user-attachments/files/16615799/BagSync.zip)

**Comment by Raenore** [comment]
> I added a new option to the Display category to show blacklisted item counts for the current player only. **Please let me know if this works for you and it's what you wanted.**
> 
> NOTE: I added a tag [Blacklist] to remind you that it's been blacklisted. [BagSync.zip](https://github.com/user-attachments/files/16615799/BagSync.zip)

Hello there,

Thank you for adding something regarding this, but unfortunately not really what I wanted/needed.

The idea was to show it just for the CURRENT CHARACTER. Not Player.
I wanted to be able to see the item count my current character had.

Showing the total count of all the characters on the player is unfortunately the same as not blacklisting it. Which, if that is what you thought I wanted, makes total sense why you thought it was weird of me asking this. :P

TL;DR: My suggestion/feedback was to add a CURRENT CHARACTER count on blacklisted items. So you could still use BagSync to know how many of a given item you have in your current bags.

Sorry for the confusion.

**Comment by Xruptor** [comment]
It's supposed to only show one character on the blacklisted item.  The active character.  If it's not doing that then it's a bug of the new feature.  I tested it myself and it worked fine.  I blacklisted some meat that multiple characters had.  I then enabled the option and it showed me the item on my active active with the count but not the others.  So if this is not working for you then it's a bug.

**Comment by Xruptor** [comment]
![image](https://github.com/user-attachments/assets/07a65358-df4e-450a-b937-4b0c93668c95)

**Comment by Raenore** [comment]
Ah, I see the issue.

I had un-blacklisted an item because of this and forgot to re-blacklist it.
I then misunderstood it since it said "Current Player" (which sounds more like all characters by a player than current character)..

But I've tested it now and it works like you've said! Thank you very much!
![image](https://github.com/user-attachments/assets/6bcefb7b-1aae-4006-99ba-b16c52f1ba58)

**Comment by Xruptor** [comment]
I'll go ahead and rename the features to Current Character from Current Player in the next update.  Be warned you may have to check the option again as it will probably default back to off with that update.
